.agents/skills: add triage-issues skill (#926)
kubernetes-sigs/agent-sandbox: .agents/skills/triage-issues/SKILL.md

| `priority/important-soon` | P1 |
| `priority/important-longterm` | P2 |
| `priority/backlog` | P3 |
| `priority/awaiting-more-evidence` | P4 |

## Key references

- Roadmap: `roadmap.md` in the repo root (source of truth for themes + status).
- Project board: https://github.com/orgs/kubernetes-sigs/projects/120 (owner `kubernetes-sigs`, number `120`).
- The board's `Priority` single-select field holds P0–P4.

## Triage heuristics

Assign each open issue a tier using, in order:

1. **Honor existing `priority/*` labels** — do not downgrade/overwrite an existing priority label unless the issue clearly changed; note any you keep.
2. **Roadmap references** — if the issue number is cited in `roadmap.md`, inherit that item's status: `⏳ In Progress` → **P1**, `📅 Planned` → **P2**, `✅ Completed` follow-ups → P2/P3.
3. **Bug severity** — data-integrity / duplicate-resource / resource-leak bugs in the core controller → **P0/P1**. Self-labeled "Critical" → P0.
4. **Value props** — latency / performance / scale bottlenecks skew **P1** (called out as a primary value proposition).
5. **Security** — security/dependency issues → **P1** (tied to "Security Fixes" roadmap item).

> 6. **Low-signal** — empty bodies, one-line questions, "collecting use cases", pinned/frozen community threads, items marked `triage/not-reproducible`/`needs-information` → **P4**. Nice-to-have features, refactors, cleanups, test scaffolding → **P3**.

Author: Fixed in d8c4d3c — changed `needs-information` to `triage/needs-information`.

> 6. **Low-signal** — empty bodies, one-line questions, "collecting use cases", pinned/frozen community threads, items marked `triage/not-reproducible`/`triage/needs-information` → **P4**. Nice-to-have features, refactors, cleanups, test scaffolding → **P3**.

Author: This looks like a false positive — both labels do exist in the repo. `gh label list --repo kubernetes-sigs/agent-sandbox` returns:

```
triage/duplicate
triage/unresolved
triage/not-reproducible
triage/needs-information
triage/accepted
```

So `triage/not-reproducible` and `triage/needs-information` (referenced at line 37) are valid and the heuristic applies as written. The verification script in the analysis appears to have captured truncated output ("Length of output: 108"), missing these entries. No change needed here.

## Procedure

1. **Verify access**: `gh auth status`; confirm repo `kubernetes-sigs/agent-sandbox`.
2. **Read the roadmap**: read `roadmap.md`; extract themes, item statuses, and any cited issue numbers.
3. **Fetch open issues**:

>    `gh issue list --repo kubernetes-sigs/agent-sandbox --state open --limit 200 --json number,title,labels,assignees,createdAt,updatedAt,body`

Author: Fixed in d8c4d3c — added `url` to the `gh issue list --json` payload.

>    `gh issue list --repo kubernetes-sigs/agent-sandbox --state open --limit 200 --json number,title,labels,assignees,createdAt,updatedAt,body,url`

Reviewer: Redirect the output to a file to avoid dumping directly into the agent's context window. Also avoids risks of exceeding tool output buffers or terminal limits. Use higher limit to be future-proof.
```suggestion
   `gh issue list --repo kubernetes-sigs/agent-sandbox --state open --limit 500 --json number,title,labels,assignees,createdAt,updatedAt,body,url > open_issues.json`
```

Author: Fixed in 3ca5bb6 — redirected output to `open_issues.json` with `--limit 500`.

4. **Discover board metadata** (IDs change per board, always re-fetch):
   - Project + Priority field/options: `gh project field-list 120 --owner kubernetes-sigs --format json` → grab the `Priority` field id and the P0–P4 option ids; project id via `gh project view 120 --owner kubernetes-sigs --format json`.

>    - Item ids: `gh project item-list 120 --owner kubernetes-sigs --limit 400 --format json` → map issue number → project item id. Add any open issue not yet on the board with `gh project item-add 120 --owner kubernetes-sigs --url <issue-url>`.

Reviewer: Project 120 already has 340 items (184 open, 156 closed). Set limit to 1000 or higher to be future-proof.
Because the triage script only edits open issues, the agent does not need item IDs for closed issues or PRs. 
```suggestion
   - Item ids: `gh project item-list 120 --owner kubernetes-sigs --limit 1000 --query "is:open" --format json` → map issue number → project item id. Add any open issue not yet on the board with `gh project item-add 120 --owner kubernetes-sigs --url <issue-url>`.
```

Author: Fixed in 3ca5bb6 — bumped limit to 1000 and filtered to open items with `--query "is:open"`.

>    - Item ids: `gh project item-list 120 --owner kubernetes-sigs --limit 1000 --query "is:open" --format json` → map issue number → project item id. Add any open issue not yet on the board with `gh project item-add 120 --owner kubernetes-sigs --url <issue-url>`.

Reviewer: When adding a new item to the project board, `gh project item-add` outputs a human-readable string by default. Adding `--format json` ensures the agent can easily parse it.

```suggestion
   - Item ids: `gh project item-list 120 --owner kubernetes-sigs --limit 1000 --query "is:open" --format json` → map issue number → project item id. Add any open issue not yet on the board with `gh project item-add 120 --owner kubernetes-sigs --url <issue-url> --format json`.
```

Author: Fixed in b7776eb — added `--format json` to the `gh project item-add` command.

5. **Triage**: assign every open issue exactly one tier using the heuristics. Sanity-check that the assigned set equals the open-issue set (no missing, no extras).
6. **Report first**: write a markdown report (`issue-triage-report.md`) grouping issues by priority with roadmap mapping + one-line rationale, and a "second look" section for judgment calls. **In an interactive run, stop and wait for the user to review before changing anything on GitHub.** In an automated/scheduled run, see below.

> 7. **Apply** (after approval / or directly in scheduled mode):

Reviewer: For clarity and safety, let's be more explicit

```suggestion
7. **Apply** (after approval in interactive mode, or filtered by untriaged issues in scheduled mode):
```

Author: Fixed in b7776eb — made the Apply step explicit about interactive (after approval) vs scheduled (filtered to untriaged issues) mode.

>    - Label: `gh issue edit <#> --repo kubernetes-sigs/agent-sandbox --add-label "<priority/...>"`. Remove any conflicting `priority/*` label if the tier changed.

Reviewer: To optimize it

```suggestion
   - Label: `gh issue edit <#> --repo kubernetes-sigs/agent-sandbox --add-label "<priority/...>" --remove-label "<priority/...>"` (if the tier changed).
```

Author: Fixed in b7776eb — label edit now uses `--add-label ... --remove-label ...` (applied when the tier changed).

   - Board field: `gh project item-edit --id <itemId> --project-id <projectId> --field-id <fieldId> --single-select-option-id <optionId>`.

>    - Drive the apply loop from a small script so all issues are processed in one pass; collect per-issue ok/err and print a final `total / errors` summary.

Reviewer: Executing API mutation calls too rapidly risks triggering GitHub's [secondary rate limits](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api#about-secondary-rate-limits).

Add a small delay (e.g., `sleep 1`) between API mutation calls in the script to throttle execution to ~300 points per minute

```suggestion
   - Drive the apply loop from a small script so all issues are processed in one pass (adding a `sleep 1` delay between API calls to respect GitHub's 900 points/min secondary rate limit); collect per-issue ok/err and print a final `total / errors` summary.
```

Author: Fixed in 3ca5bb6 — added `sleep 1` between API calls in the apply loop.

## Procedure

1. **Verify access**: `gh auth status`; confirm repo `kubernetes-sigs/agent-sandbox`.
2. **Read the roadmap**: read `roadmap.md`; extract themes, item statuses, and any cited issue numbers.
3. **Fetch open issues**:
   `gh issue list --repo kubernetes-sigs/agent-sandbox --state open --limit 500 --json number,title,labels,assignees,createdAt,updatedAt,body,url > open_issues.json`
4. **Discover board metadata** (IDs change per board, always re-fetch):
   - Project + Priority field/options: `gh project field-list 120 --owner kubernetes-sigs --format json` → grab the `Priority` field id and the P0–P4 option ids; project id via `gh project view 120 --owner kubernetes-sigs --format json`.
   - Item ids: `gh project item-list 120 --owner kubernetes-sigs --limit 1000 --query "is:open" --format json` → map issue number → project item id. Add any open issue not yet on the board with `gh project item-add 120 --owner kubernetes-sigs --url <issue-url> --format json`.
5. **Triage**: assign every open issue exactly one tier using the heuristics. Sanity-check that the assigned set equals the open-issue set (no missing, no extras).

> 6. **Report first**: write a markdown report (`issue-triage-report.md`) grouping issues by priority with roadmap mapping + one-line rationale, and a "second look" section for judgment calls. **In an interactive run, stop and wait for the user to review before changing anything on GitHub.** In an automated/scheduled run, see below.

Reviewer: To ensure consistent reporting across different agent runs, let's clarify what constitutes a judgment call that warrants inclusion in the second look section.

```suggestion
6. **Report first**: write a markdown report (`issue-triage-report.md`) grouping issues by priority with roadmap mapping + one-line rationale, and a "second look" section for judgment calls (e.g., heuristic conflicts, boundary cases, insufficient info, or ambiguous roadmap items). **In an interactive run, stop and wait for the user to review before changing anything on GitHub.** In an automated/scheduled run, see below.
```

Author: Done in 2aff7be — added examples (heuristic conflicts, boundary cases, insufficient info, ambiguous roadmap items) so the "second look" section is reported consistently across runs.
